JSON exception while parsing item NBT data

Triton v4  
Purpur 1.19.2

Plugin: https://www.spigotmc.org/resources/itemjoin.12661/

![image](https://user-images.githubusercontent.com/7467891/191560937-586ff067-733a-42ba-bcf5-b7b7ed3faf7b.png)


Stacktrace:
```java
[16:26:59] [Server thread/ERROR]: [Triton] Unhandled exception occurred in onPacketSending(PacketEvent) for Triton
com.google.gson.JsonSyntaxException: java.io.EOFException: End of input at line 1 column 2 path $.
	at com.google.gson.Gson.fromJson(Gson.java:973) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:928) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:877) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:848) ~[gson-2.8.9.jar:?]
	at net.kyori.adventure.text.serializer.gson.GsonComponentSerializerImpl.deserialize(GsonComponentSerializerImpl.java:84) ~[adventure-text-serializer-gson-4.11.0.jar:?]
	at net.kyori.adventure.text.serializer.gson.GsonComponentSerializerImpl.deserialize(GsonComponentSerializerImpl.java:37) ~[adventure-text-serializer-gson-4.11.0.jar:?]
	at com.rexcantor64.triton.utils.ComponentUtils.deserializeFromJson(ComponentUtils.java:27) ~[Triton-Beta.jar:?]
	at com.rexcantor64.triton.spigot.utils.ItemStackTranslationUtils.translateNbtItem(ItemStackTranslationUtils.java:169) ~[Triton-Beta.jar:?]
	at com.rexcantor64.triton.spigot.utils.ItemStackTranslationUtils.translateItemStack(ItemStackTranslationUtils.java:87) ~[Triton-Beta.jar:?]
	at com.rexcantor64.triton.spigot.packetinterceptor.ProtocolLibListener.handleSetSlot(ProtocolLibListener.java:531) ~[Triton-Beta.jar:?]
	at com.rexcantor64.triton.spigot.packetinterceptor.ProtocolLibListener.onPacketSending(ProtocolLibListener.java:738) ~[Triton-Beta.jar:?]
	at com.comphenix.protocol.injector.SortedPacketListenerList.invokeSendingListener(SortedPacketListenerList.java:195) ~[ProtocolLib-LATEST-[600].jar:?]
	at com.comphenix.protocol.injector.SortedPacketListenerList.invokePacketSending(SortedPacketListenerList.java:149) ~[ProtocolLib-LATEST-[600].jar:?]
	at com.comphenix.protocol.injector.PacketFilterManager.postPacketToListeners(PacketFilterManager.java:547) ~[ProtocolLib-LATEST-[600].jar:?]
	at com.comphenix.protocol.injector.PacketFilterManager.invokePacketSending(PacketFilterManager.java:521) ~[ProtocolLib-LATEST-[600].jar:?]
	at com.comphenix.protocol.injector.netty.manager.NetworkManagerInjector.onPacketSending(NetworkManagerInjector.java:99) ~[ProtocolLib-LATEST-[600].jar:?]
	at com.comphenix.protocol.injector.netty.channel.NettyChannelInjector.processOutbound(NettyChannelInjector.java:570) ~[ProtocolLib-LATEST-[600].jar:?]
	at com.comphenix.protocol.injector.netty.channel.NettyChannelInjector$2.proxyRunnable(NettyChannelInjector.java:473) ~[ProtocolLib-LATEST-[600].jar:?]
	at com.comphenix.protocol.injector.netty.channel.NettyEventLoopProxy.execute(NettyEventLoopProxy.java:220) ~[ProtocolLib-LATEST-[600].jar:?]
	at net.minecraft.network.Connection.sendPacket(Connection.java:432) ~[?:?]
	at net.minecraft.network.Connection.processQueue(Connection.java:540) ~[?:?]
	at net.minecraft.network.Connection.flushQueue(Connection.java:503) ~[?:?]
	at net.minecraft.network.Connection.send(Connection.java:398) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.send(ServerGamePacketListenerImpl.java:2256) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.send(ServerGamePacketListenerImpl.java:2242) ~[?:?]
	at net.minecraft.server.level.ServerPlayer$1.sendSlotChange(ServerPlayer.java:296) ~[?:?]
	at net.minecraft.world.inventory.AbstractContainerMenu.synchronizeSlotToRemote(AbstractContainerMenu.java:320) ~[?:?]
	at net.minecraft.world.inventory.AbstractContainerMenu.broadcastChanges(AbstractContainerMenu.java:243) ~[?:?]
	at net.minecraft.server.level.ServerPlayer.tick(ServerPlayer.java:650) ~[?:?]
	at net.minecraft.server.level.ServerLevel.tickNonPassenger(ServerLevel.java:1242) ~[?:?]
	at net.minecraft.server.level.ServerLevel.lambda$tick$6(ServerLevel.java:721) ~[?:?]
	at net.minecraft.world.level.entity.EntityTickList.forEach(EntityTickList.java:42) ~[purpur-1.19.2.jar:git-Purpur-1783]
	at net.minecraft.server.level.ServerLevel.tick(ServerLevel.java:699) ~[?:?]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1582) ~[purpur-1.19.2.jar:git-Purpur-1783]
	at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:486) ~[purpur-1.19.2.jar:git-Purpur-1783]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1444) ~[purpur-1.19.2.jar:git-Purpur-1783]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1214) ~[purpur-1.19.2.jar:git-Purpur-1783]
	at net.minecraft.server.MinecraftServer.lambda$spin$1(MinecraftServer.java:310) ~[purpur-1.19.2.jar:git-Purpur-1783]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.io.EOFException: End of input at line 1 column 2 path $.
	at com.google.gson.stream.JsonReader.nextNonWhitespace(JsonReader.java:1395) ~[gson-2.8.9.jar:?]
	at com.google.gson.stream.JsonReader.doPeek(JsonReader.java:493) ~[gson-2.8.9.jar:?]
	at com.google.gson.stream.JsonReader.hasNext(JsonReader.java:413) ~[gson-2.8.9.jar:?]
	at net.kyori.adventure.text.serializer.gson.ComponentSerializerImpl.read(ComponentSerializerImpl.java:132) ~[adventure-text-serializer-gson-4.11.0.jar:?]
	at net.kyori.adventure.text.serializer.gson.ComponentSerializerImpl.read(ComponentSerializerImpl.java:56) ~[adventure-text-serializer-gson-4.11.0.jar:?]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:963) ~[gson-2.8.9.jar:?]
	... 38 more
[16:26:59] [Server thread/ERROR]: Parameters: 
  net.minecraft.network.protocol.game.PacketPlayOutSetSlot@11c00fee[
    c=0
    d=8
    e=40
    f=1 nether_star
  ]
```

--- a/triton-spigot/src/main/java/com/rexcantor64/triton/spigot/utils/ItemStackTranslationUtils.java
+++ b/triton-spigot/src/main/java/com/rexcantor64/triton/spigot/utils/ItemStackTranslationUtils.java
@@ -8,15 +8,11 @@ import com.comphenix.protocol.wrappers.nbt.NbtBase;
 import com.comphenix.protocol.wrappers.nbt.NbtCompound;
 import com.comphenix.protocol.wrappers.nbt.NbtFactory;
 import com.comphenix.protocol.wrappers.nbt.NbtList;
+import com.google.gson.JsonSyntaxException;
 import com.rexcantor64.triton.api.language.Localized;
-import com.rexcantor64.triton.config.MainConfig;
 import com.rexcantor64.triton.spigot.SpigotTriton;
 import com.rexcantor64.triton.utils.ComponentUtils;
 import lombok.val;
-import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
-import net.md_5.bungee.api.chat.BaseComponent;
-import net.md_5.bungee.api.chat.TextComponent;
-import net.md_5.bungee.chat.ComponentSerializer;
 import org.bukkit.Material;
 import org.bukkit.inventory.ItemStack;
 import org.bukkit.inventory.meta.ItemMeta;
@@ -92,19 +88,7 @@ public class ItemStackTranslationUtils {
                     Collection<NbtBase<String>> pagesCollection = pages.asCollection();
                     List<String> newPagesCollection = new ArrayList<>();
                     for (NbtBase<String> page : pagesCollection) {
-                        if (page.getValue().startsWith("\"")) {
-                            val pageString = page.getValue().substring(1, page.getValue().length() - 1);
-                            val pageComponent = LegacyComponentSerializer.legacySection().deserialize(pageString);
-                            main().getMessageParser()
-                                    .translateComponent(
-                                            pageComponent,
-                                            languagePlayer,
-                                            main().getConfig().getItemsSyntax()
-                                    )
-                                    .map(ComponentUtils::serializeToJson)
-                                    .ifChanged(newPagesCollection::add)
-                                    .ifUnchanged(() -> newPagesCollection.add(ComponentUtils.serializeToJson(pageComponent)));
-                        } else {
+                        try {
                             main().getMessageParser()
                                     .translateComponent(
                                             ComponentUtils.deserializeFromJson(page.getValue()),
@@ -114,6 +98,16 @@ public class ItemStackTranslationUtils {
                                     .map(ComponentUtils::serializeToJson)
                                     .ifChanged(newPagesCollection::add)
                                     .ifUnchanged(() -> newPagesCollection.add(page.getValue()));
+                        } catch (JsonSyntaxException e) {
+                            val pageString = page.getValue();
+                            main().getMessageParser()
+                                    .translateString(
+                                            pageString,
+                                            languagePlayer,
+                                            main().getConfig().getItemsSyntax()
+                                    )
+                                    .ifChanged(newPagesCollection::add)
+                                    .ifUnchanged(() -> newPagesCollection.add(pageString));
                         }
                     }
                     compound.put("pages", NbtFactory.ofList("pages", newPagesCollection));
@@ -163,7 +157,7 @@ public class ItemStackTranslationUtils {
         NbtCompound display = compound.getCompoundOrDefault("display");
         if (display.containsKey("Name")) {
             String name = display.getStringOrDefault("Name");
-            if (main().getMcVersion() >= 13) {
+            try {
                 main().getMessageParser()
                         .translateComponent(
                                 ComponentUtils.deserializeFromJson(name),
@@ -174,7 +168,7 @@ public class ItemStackTranslationUtils {
                         .map(ComponentUtils::serializeToJson)
                         .ifChanged(result -> display.put("Name", result))
                         .ifToRemove(() -> display.remove("Name"));
-            } else {
+            } catch (JsonSyntaxException e) {
                 main().getMessageParser()
                         .translateString(
                                 name,
@@ -183,6 +177,7 @@ public class ItemStackTranslationUtils {
                         )
                         .ifChanged(result -> display.put("Name", result))
                         .ifToRemove(() -> display.remove("Name"));
+
             }
         }
 
@@ -191,7 +186,7 @@ public class ItemStackTranslationUtils {
 
             List<String> newLore = new ArrayList<>();
             for (String lore : loreNbt) {
-                if (main().getMcVersion() >= 13) {
+                try {
                     main().getMessageParser()
                             .translateComponent(
                                     ComponentUtils.deserializeFromJson(lore),
@@ -206,7 +201,7 @@ public class ItemStackTranslationUtils {
                                             .collect(Collectors.toList())
                             ))
                             .ifUnchanged(() -> newLore.add(lore));
-                } else {
+                } catch (JsonSyntaxException e) {
                     main().getMessageParser()
                             .translateString(
                                     lore,

